packages/mobile/src/pincode/PincodeSet.tsx

```diff
@@ -111,7 +137,20 @@ export class PincodeSet extends React.Component<Props, State> {
       this.props.setPincode(PincodeType.CustomPin)
       ValoraAnalytics.track(OnboardingEvents.pin_set)
       this.navigateToNextScreen()
+      if (this.isChangingPin()) {
+        const updated = await updatePin(this.props.account, this.state.oldPin, pin2)
```

Comment:
Can you just create an `oldPin` variable here using `getCachedPin(DEFAULT_CACHE_ACCOUNT) ?? '' })` and pass it in directly instead of storing to state?

Reply:
The reason not to do that is that the `getCachedPin` might expire in the time between the user enters their PIN the first time and they enter the new one. The cache lasts for 5 minutes, so it's unlikely to happen, but if it did the user would be asked for their old password again after inputting the new one, which will likely be pretty confusing.

I'll add a comment to the code since it's not very clear.

Reply:
Thinking it over, if a user takes more than 5 minutes to input their new PIN maybe we want to expire their request or something and have them start over? For security reasons?

Reply:
Oh I see. I don't really think there's much of a security concern...probably not worth the effort for an unlikely edge case.

Another option is to make a property on the component that holds this value. Whenever I see something in component state I expect that it will be changed and trigger re-renders so it was confusing for me to see it there. In a functional component we'd define this with `const` instead of `useState` right?

Reply:
In a functional component I'd probably use a `useMemo` hook instead. Using just a const would re-fetch from the cache in every render, but the useMemo would cache the result. Not sure if there's an easy equivalent for class components :/